pkg/service/record/record.go

```diff
@@ -312,3 +325,64 @@ func (r *Recorder) StartMock(ctx context.Context) error {
 	utils.LogError(r.logger, err, stopReason)
 	return fmt.Errorf(stopReason)
 }
+
+func (r *Recorder) ReRecord(ctx context.Context) error {
+
+	tcs, err := r.testDB.GetTestCases(ctx, r.config.ReRecord)
+	if err != nil {
+		return err
+	}
+
+	allTestCasesRecorded := true
+	for _, tc := range tcs {
+		host, port, err := extractHostAndPort(tc.Curl)
+		if err != nil {
+			r.logger.Error("Failed to extract host and port", zap.Error(err))
+			allTestCasesRecorded = false
+			continue // Proceed with the next command
+		}
+
+		if err := waitForPort(ctx, host, port); err != nil {
```

Comment:
Why we need to wait for port before each simulate request. Cant we move it outside the for loop of testcases ??